hooks/create_move/movement.c

```diff
@@ -0,0 +1,54 @@
+#include "../../config/config.h"
+#include "../../utils/utils.h"
+#include "../../source_sdk/entity/entity.h"
+#include "../../source_sdk/math/vec3.h"
+#include "../../source_sdk/user_cmd.h"
+#include "../paint_traverse/paint_traverse.h"
+#include "create_move.h"
+
+void bhop(void *localplayer, struct user_cmd *user_cmd){
+
+    /* scouts double jump only works sometimes, very inconsistent
+     * i haven't figured out why (FROM TESTING: i belive the on ground flag sometimes is not getting set correctly)
+     * in any case, i'll just disable bhop on scout until i figure it out */
+
+    if(get_player_class(localplayer) == TF_CLASS_SCOUT){
+        return;
+    }
+
+    static bool was_jumping = false;
+    bool on_ground = (get_ent_flags(localplayer) & FL_ONGROUND);
+
+    if(user_cmd->buttons & IN_JUMP){
+        if(!was_jumping && !on_ground){
+            user_cmd->buttons &= ~IN_JUMP;
+        }
+        else if(was_jumping){
+            was_jumping = false;
+        }
+    } else if(!was_jumping){
+        was_jumping = true;
+    }
+}
+
+void autostrafe(void *localplayer, struct user_cmd *user_cmd){
+
+    // TBD: figure out how to implement directional / rage autostrafe
+    // i don't think this kind of autostrafe does anything
```

Comment:
Clarifying: i don't think its very useful, since you are mostly manually strafing for it.

Reply:
Agreed rage strafe is far better but this is nice for now and can be easily disabled